docs/tutorials/data_provider.md

```diff
@@ -62,6 +62,67 @@ impl AdditiveIdentity {
 }
 ```
 
+## Loading Additional Data at Runtime
+
+A key feature of ICU4X is the ability to download data dynamically, allowing clients to load additional locales at runtime.
+
+Dynamic data loading can currently be performed in user code. A future core library API may provide this functionality; please submit feedback in [#2985](https://github.com/unicode-org/icu4x/issues/2985).
+
+```rust
+use icu_provider_adapters::either::EitherProvider;
+use icu_provider_adapters::fallback::LocaleFallbackProvider;
+use icu_provider_adapters::fork::ForkByKeyProvider;
+use icu_provider_adapters::fork::MultiForkByErrorProvider;
+use icu_provider_adapters::fork::predicates::MissingLocalePredicate;
+use icu_provider_blob::BlobDataProvider;
+use icu_provider_fs::FsDataProvider;
+use icu_provider::DataLocale;
+use icu_provider::hello_world::HelloWorldFormatter;
+use icu::locid::locale;
+use icu::locid::subtags::Language;
+use std::path::Path;
+use writeable::Writeable;
+
+// Create the empty MultiForkByErrorProvider:
+let mut provider = MultiForkByErrorProvider::new_with_predicate(
+    vec![],
+    MissingLocalePredicate
+);
+
+// Helper function to add data into the growable provider on demand:
+let mut get_hello_world_formatter = |loc: &DataLocale| {
+    // Try to create the formatter a first time with data that has already been loaded.
+    if let Ok(formatter) = HelloWorldFormatter::try_new_with_buffer_provider(&provider, loc) {
+        return formatter;
+    }
+
+    // We failed to create the formatter. Load more data for the language.
+    let path_buf = 
+        Path::new("../../provider/adapters/tests/data/langtest")
+        .join(&*loc.language().as_str());
+    let lang_provider = match FsDataProvider::try_new(&path_buf) {
```

Comment:
thought: I think `FsDataProvider` isn't a great example, because it's basically free to construct, and using a single `FsDataProvider` can achieve the same thing. It would be more educational to load a blob into memory.

Reply:
I used JSON because (1) this data is already being built for another test, (2) when in doubt, JSON in the repo is better than binary postcard files in the repo, and (3) tutorial readers can go and look and see exactly what data is being loaded; they can't do that with an opaque postcard file.

Reply:
JsonBlobProvider when 😆